ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/permIdRequest/presenter/PermanentIdRequestPresenter.java

```diff
@@ -0,0 +1,156 @@
+import org.iplantc.de.resources.client.messages.I18N;
+
+import com.google.gwt.user.client.rpc.AsyncCallback;
+import com.google.gwt.user.client.ui.HasOneWidget;
+import com.google.inject.Inject;
+import com.google.web.bindery.autobean.shared.AutoBean;
+import com.google.web.bindery.autobean.shared.AutoBeanCodex;
+
+/**
+ * 
+ * 
+ * @author sriram
+ * 
+ */
+public class PermanentIdRequestPresenter implements Presenter {
+
+    PermanentIdRequestView view;
+
+    final DiskResourceServiceFacade drsvc;
+
+    final PermanentIdRequestAdminServiceFacade prsvc;
+
+    private PermanentIdRequest selectedRequest;
+
+    final DiskResourceUtil diskResourceUtil = DiskResourceUtil.getInstance();
+
+    private final PermanentIdRequestAutoBeanFactory factory;
+
+    private final PermanentIdRequestPresenterAppearance appearance;
+
+    @Inject
+    public PermanentIdRequestPresenter(DiskResourceServiceFacade drsvc,
+                                       PermanentIdRequestAdminServiceFacade prsvc,
+                                       PermanentIdRequestAutoBeanFactory factory,
+                                       PermanentIdRequestView view,
+                                       PermanentIdRequestPresenterAppearance appearance) {
+        this.drsvc = drsvc;
+        this.prsvc = prsvc;
+        this.view = view;
+        this.factory = factory;
+        this.appearance = appearance;
+        view.setPresenter(this);
+    }
+
+    @Override
+    public void fetchMetadata() {
+        view.fetchMetadata(selectedRequest.getFolder(), appearance, drsvc);
+    }
+
+    @Override
+    public void setSelectedRequest(PermanentIdRequest request) {
+        this.selectedRequest = request;
+    }
+
+    @Override
+    public void go(HasOneWidget container) {
+        container.setWidget(view);
+        loadPermIdRequests();
+
+    }
+
+    @Override
+    public void getPermIdRequests() {
+        prsvc.getPermanentIdRequests(new AsyncCallback<String>() {
+
+            @Override
+            public void onFailure(Throwable caught) {
+                IplantAnnouncer.getInstance()
+                               .schedule(new ErrorAnnouncementConfig(appearance.requestLoadFailure()));
+            }
+
+            @Override
+            public void onSuccess(String result) {
+                final AutoBean<PermanentIdRequestList> decode = AutoBeanCodex.decode(factory,
+                                                                                     PermanentIdRequestList.class,
+                                                                                     result);
+
+                view.loadRequests(decode.as().getRequests());
+            }
+        });
+
+    }
+
+    @Override
+    public void loadPermIdRequests() {
+        getPermIdRequests();
+    }
+
+    @Override
+    public void updateRequest(final PermanentIdRequestUpdate update) {
+        view.mask(I18N.DISPLAY.loadingMask());
+        if (selectedRequest != null && update != null) {
```

Comment:
This mask should also be nested inside this `if` check, otherwise if the check fails, the view will never unmask.